### ISSUE TITLE
Are there any planning to abstract ui and core functions in separate projects?

### ISSUE THREAD

**Issue by codeTheWorld69** [LagradOst/CloudStream-3#1374]
### Describe your suggested feature

If there is any planning to abstract core logic into separate repo then that can be used as a backend to make frontend projects in flutter or any other cross-platform tool

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Blatzar** [comment]
Yes, we've already made networking a [library](https://github.com/Blatzar/NiceHttp), working on [downloading](https://github.com/Blatzar/FetchButton) and we have plans to make all providers in a separate repo utilizing dex to instantly update providers without the need for user interaction in the app.

The current libraries are not very cross compatible, but we'll be able to easily fix that once we get there.
